core/src/authorizations.ts

```diff
@@ -24,14 +30,26 @@ export const authorizeCollateral = async function (
     return executeTransaction(network, 'MCD_VAT', contractMethod, [clipperAddress], notifier);
 };
 
-export const getWalletAuthorizationStatus = async function (network: string, walletAddress: string): Promise<boolean> {
+export const authorizeCollateral = memoizee(_authorizeCollateral, {
+    maxAge: AUTHORIZE_WALLET_CACHE_MS,
+    promise: true,
+    length: 4,
+});
+
+const _getWalletAuthorizationStatus = async function (network: string, walletAddress: string): Promise<boolean> {
     const joinDaiAddress = await getContractAddressByName(network, 'MCD_JOIN_DAI');
     const contract = await getContract(network, 'MCD_VAT');
     const authorizationStatus = await contract.can(walletAddress, joinDaiAddress);
     return authorizationStatus.toNumber() === 1;
 };
 
-export const getCollateralAuthorizationStatus = async function (
+export const getWalletAuthorizationStatus = memoizee(_getWalletAuthorizationStatus, {
+    maxAge: GET_AUTHORIZATION_STATUS_CACHE_MS,
+    promise: true,
+    length: 2,
+});
```

Comment:
As discussed, you can not cache "get" methods, otherwise they will give misleading results as you will call authorization in many times (especially since you've caching it for `maxAge`)

(or maybe there is a way to invalidate cache that you can call from the `execute*` methods?)

Reply:
> (or maybe there is a way to invalidate cache that you can call from the `execute*` methods?)

I can call `memoized.delete("foo", true)` to clear a certain cache. I could see this working if I call it before calling the function in the `execute*` methods.

Reply:
I just noticed that this probably wouldn't work, as the calling of the function happens in another project. I don't know if I can clear the cache from one project in another project.

Reply:
> as the calling of the function happens in another project

Not really, you only need to invalidate cache once you authorized, which happens in the same file.